src/python/WMCore/Storage/RucioFileCatalog.py

```diff
@@ -0,0 +1,231 @@
+    """
+    #get site config
+    siteConfigPath = os.getenv('SITECONFIG_PATH',None)
+    if not siteConfigPath:
+        raise RuntimeError('SITECONFIG_PATH is not defined')
+    subPath = ''
+    #the storage site is where jobs are executed so use local path given in SITECONFIG_PATH to locate storage.json
+    if currentSite == storageSite:
+        #it is a site (no defined subSite), storage.json is located at the path given in SITECONFIG_PATH
+        if currentSubsite is None:
+            subPath = siteConfigPath
+        #it is a subsite, move one level up
+        else:
+            subPath = siteConfigPath + '/..'
+    #cross site
+    else:
+        #it is a site (no defined subSite), move one level up
+        if currentSubsite is None:
+            subPath = siteConfigPath + '/../' + storageSite
+        #it is a subsite, move two levels up
+        else:
+            subPath = siteConfigPath + '/../../' + storageSite
+    pathToStorageDescription = subPath + '/storage.json'
+    pathToStorageDescription = os.path.normpath(os.path.realpath(pathToStorageDescription))#resolve symbolic link and relative path?
+    return pathToStorageDescription 
+
+def readRFC(filename, storageSite, volume, protocol):
+    """
+    _readRFC_
+
+    Read the provided storage.json and return a RucioFileCatalog
+    instance containing the details found in it
+    """
+    rfcInstance = RucioFileCatalog()
+    try:
+        jsonFile = open(filename,encoding="utf-8")
+        jsElements = json.load(jsonFile)
+    except Exception as ex:
+        msg = "Error reading storage description file: %s\n" % filename
+        msg += str(ex)
+        raise RuntimeError(msg)
+    #now loop over elements, select the right one and fill lfn-to-pfn
+    for jsElement in jsElements:
+        #check to see if the storageSite and volume matchs with "site" and "volume" in storage.json
+        if jsElement['site'] == storageSite and jsElement['volume'] == volume: 
+            #now loop over protocols
+            #!!!!!!!!!!!!!!Maybe save all protocols so that chain works
+            for proc in jsElement['protocols']:
+                #check found match
+                if proc['protocol'] == protocol:
+                    rfcInstance.preferredProtocol = protocol
+                    chain = proc['chain'] if 'chain' in proc.keys() else None
+                    #check if prefix is in protocol block
+                    if 'prefix' in proc.keys():
+                        #lfn-to-pfn
+                        match = '(.*)' #match all
+                        result = proc['prefix']+'/$1'
+                        rfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                        #pfn-to-lfn
+                        match = proc['prefix']+'/(.*)'
+                        result = '/$1'
+                        rfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
```

Comment:
Why is the need to create additional regular expression mappings back and forth from `lfn-to-pfn` and `pfn-to-lfn`  here,  similar to what we were doing before. I though, one of the benefits of having `storage.json` as it is configured now, should be the ability to get rid of those regular expression operations and blindly use what ever is provided in the `prefix` field. Of course I am talking about the case where no explicit mapping rules are defined for the protocol under consideration. For those which there are already some rules and no `prefix` field, of course for them, there is no way to avoid that.

Reply:
As far as I understand there are cases of chained rules where you can not use `prefix`, it must be regular expression to resolve kind of nested paths
https://gitlab.cern.ch/SITECONF/T1_DE_KIT/-/blob/master/storage.json?ref_type=heads#L64